cmd/lakefs/cmd/setup.go

```diff
@@ -32,14 +32,24 @@ var setupCmd = &cobra.Command{
 		ctx := cmd.Context()
 		dbParams := cfg.GetDatabaseParams()
 
+		if len(dbParams.Type) > 0 && len(dbParams.ConnectionString) > 0 { // Conflicting configuration
+			fmt.Printf("Conflicting database parameters, connection_string should be defined for the specific driver. Do you need to go through migration?\n")
+			os.Exit(1)
+		}
 		var (
 			dbPool   db.Database
 			kvParams params.KV
 			kvStore  kv.Store
 			migrator db.Migrator
+			err      error
```

Comment:
Not crucial and mainley a matter of personal preference, but I would remove the `err` declaration from here and use the short declaration in both sections below.
For me, as a code reader, declaring a `var` implies that it has some wider relevancy, like the `migrator` that is initialized withinh the `if/else` block and is used afterwards, while the `error` is only used locally and whatever value it gets, is meaningless immediately after the check.
Again, that's my personal opinion :)

Reply:
It's an issue since doing that Go treats the internal kvStore as a new parameter in the initilization and passes a nil pointer afterwards